src/cc/libclient/Monitor.cc

```diff
@@ -309,51 +307,67 @@ class Monitor::Impl: public QCRunnable {
                     else {
                         Properties* clientCounters = clientPtr->GetStats();
                         AggregateCounters(*clientCounters,
-                                sumOfAliveClientCounters);
+                                newSumOfClientCounters);
                         KfsClient::DisposeProperties(clientCounters);
                         ++clientInd;
                     }
                 }
                 theClientRemovalLock.Unlock();
                 fsMetrics.clients.erase(fsMetrics.clients.begin()+numClients,
                         fsMetrics.clients.end());
-                ClientCounters* sumOfAllClientCounters;
-                if (sumOfAliveClientCounters.empty()) {
-                    // All clients have been removed. The final sum is equal to
-                    // to the sum of removed clients.
-                    sumOfAllClientCounters = &sumOfRemovedClientCounters;
-                }
-                else if (sumOfRemovedClientCounters.empty()) {
-                    // No clients have been removed. The final sum is equal to
-                    // the sum of alive clients.
-                    sumOfAllClientCounters = &sumOfAliveClientCounters;
-                }
-                else {
-                    // We have both alive and removed clients. The final sum is
-                    // equal to the sum of both.
-                    if (sumOfRemovedClientCounters.size() !=
-                            sumOfAliveClientCounters.size()) {
-                        // For each counter in sumOfAliveClientCounters,
-                        // there must be a matching counter in
-                        // fsMetrics.sumOfRemovedClientCounters.
-                        KFS_LOG_STREAM_ERROR
-                            << "Monitor: Number of counters "
-                            << "in alive and removed clients don't match!"
-                            << KFS_LOG_EOM;
+
+                if (!fsMetrics.currSumOfClientCounters.empty()) {
```

Comment:
Error check should be done only for second report and onwards. First time we report counters, the current sum of client counters should be empty.


-------- Original message --------
From: Faraaz Sareshwala <notifications@github.com>
Date: 4/18/16 4:47 PM (GMT-08:00)
To: quantcast/qfs <qfs@noreply.github.com>
Cc: Mehmet CanKurt <mcanKurt@quantcast.com>
Subject: Re: [quantcast/qfs] Topic/qfs client monitor (#126)


In src/cc/libclient/Monitor.cc<https://github.com/quantcast/qfs/pull/126#discussion_r60153923>:

> -                    sumOfAllClientCounters = &sumOfAliveClientCounters;
> -                }
> -                else {
> -                    // We have both alive and removed clients. The final sum is
> -                    // equal to the sum of both.
> -                    if (sumOfRemovedClientCounters.size() !=
> -                            sumOfAliveClientCounters.size()) {
> -                        // For each counter in sumOfAliveClientCounters,
> -                        // there must be a matching counter in
> -                        // fsMetrics.sumOfRemovedClientCounters.
> -                        KFS_LOG_STREAM_ERROR
> -                            << "Monitor: Number of counters "
> -                            << "in alive and removed clients don't match!"
> -                            << KFS_LOG_EOM;
> +
> +                if (!fsMetrics.currSumOfClientCounters.empty()) {


I would check for the error and return to make the code as less nested as possible.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/quantcast/qfs/pull/126/files/daf7e0c991ee0fd59aeae919fc271683e46aa6cd#r60153923>